src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -8366,6 +8366,12 @@ void System.Collections.ICollection.CopyTo(System.Array array, int index) { }
         void System.Collections.IList.Insert(int index, object? value) { }
         void System.Collections.IList.Remove(object? value) { }
     }
+    public static partial class ReadOnlyCollection
+    {
+        public static System.Collections.ObjectModel.ReadOnlyCollection<T> CreateCollection<T>(params System.ReadOnlySpan<T> values) { throw null; }
```

Comment:
We should have `[EditorBrowsable(Never)]` on both these `Create` methods (same as the approved API surface in https://github.com/dotnet/runtime/issues/108457#issuecomment-2400438835).

Reply:
> We should have `[EditorBrowsable(Never)]` on both these `Create` methods (same as the approved API surface in [#108457 (comment)](https://github.com/dotnet/runtime/issues/108457#issuecomment-2400438835)).

It was here but @stephentoub wrote ["This should not be marked with EditorBrowsable"](https://github.com/dotnet/runtime/pull/111093/commits/74f8733b81974cd242898279346e038936fddf21#r1903262956)

[API review](https://github.com/dotnet/runtime/issues/110161#issuecomment-2515391504) do not have this attribute also.

Should I return `[EditorBrowsable(Never)]` attribute to both methods?

Reply:
I thought API review had just left those out accidentally. I don't feel strongly either way, but we should make sure to be consistent with the other factory APIs and any other ones we're adding I assume (eg. the ones in the other issue).

Reply:
No, it wasn't accidental. These are perfectly valid for a dev to call directly. Please remove them.

Reply:
If those methods are meant to be called directly then based on this [#110161 (comment)](https://github.com/dotnet/runtime/issues/110161#issuecomment-2515391504) it seams more natural to create 1 non-generic type per collection
like it is done for Immutable and Frozen collections.

Reply:
> If those methods are meant to be called directly then based on this [#110161 (comment)](https://github.com/dotnet/runtime/issues/110161#issuecomment-2515391504) it seams more natural to create 1 non-generic type per collection like it is done for Immutable and Frozen collections.

I'll create two different classes for these methods like for the Immutable and Frozen collections, and remove the [EditorBrowsable(Never)] attribute.

Reply:
Pretty sure this would have to go back to API review to split it into different classes, as that's not the API shape that was approved.

As for the attribute, if we're removing it, should we also remove it from the approved API surface of the other `Create` methods for the other collection types? Or is there a specific reason only those are hidden?

Reply:
> As for the attribute, if we're removing it, should we also remove it from the approved API surface of the other Create methods for the other collection types?

Which are you referring to?

Reply:
I separated the methods into different classes, renamed them as `Create`, and removed the [EditorBrowsable(Never)] attribute.

Reply:
> Which are you referring to?

I mean the ones on the factory methods approved in https://github.com/dotnet/runtime/issues/108457#issuecomment-2400438835.

Reply:
> > Which are you referring to?
> 
> I mean the ones on the factory methods approved in [#108457 (comment)](https://github.com/dotnet/runtime/issues/108457#issuecomment-2400438835).

Those are marked "needs work", with the comment "Rather than putting these on CollectionExtensions, we should just add public ctors that accept ReadOnlySpan"

Reply:
> Pretty sure this would have to go back to API review to split it into different classes, as that's not the API shape that was approved.

@Sergio0694 API review says
 
* We could add another type (e.g. System.Collections.ObjectModel.Collection **or one non-generic type per collection (like we do in immutable)**.

Were two classes also approved?

Reply:
> Were two classes also approved?

No, these are notes about what got discussed during review. The API as approved puts both factories under the `ReadOnlyCollection` class. Could you update your changes to reflect that? Thanks.

Reply:
I'll move it back to one class.